lib/github_changelog_generator/generator/generator_fetcher.rb

```diff
@@ -96,6 +99,16 @@ def associate_tagged_prs(tags, prs, total)
       end
     end
 
+    def try_merge_commit_sha_from_gitlab(merge_request)
```

Comment:
The naming here has issues:

- `merge_request` argument name - it can be a PR _or_ a Merge Request (e.g. `pr_or_merge_request` is not an ill-descriptive name for that)
- the method name `#try_merge_commit_sha_from_gitlab` obscures that it _also_ processes GitHub data 

A method name such as `#find_merge_commit_for(pr_or_merge_request)` would help maintenance of this.